plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java

```diff
@@ -277,6 +278,44 @@ public boolean intersects(int start, int end) {
                 return (start <= offset && end >= offset) || (start <= endOffset && end >= endOffset)
                         || (start >= offset && end <= endOffset);
             }
+            
+            @Override
+            public int hashCode() {
```

Comment:
Sorry for drive-by commenting, just saw this change and wanted to suggest using `Objects.hash()` (and maybe Objects.equals() in the following equals() method). We use them a lot in e.g the query builders and it makes these methods much more friendly to read IMHO.